components/hab/src/cli.rs

```diff
@@ -535,6 +535,8 @@ pub fn get() -> App<'static, 'static> {
                     environment variable if defined. (default: https://bldr.habitat.sh)")
                 (@arg PKG_IDENT: +required +takes_value {valid_fully_qualified_ident} "A fully qualified package identifier \
                     (ex: core/busybox-static/1.42.2/20170513215502)")
+                (@arg PKG_TARGET: +takes_value {valid_target}
+                    "A package target (ex: x86_64-windows) (default: x86_64-linux)")
```

Comment:
Shouldn't the default be whatever platform the `hab` executable you're using to run this command is compiled for?

I can imagine that with more people using Habitat on Windows, it'll quickly get old having to constantly specify `x86_64-windows` on everything.

(I know that's not part of this change, strictly speaking, but it feels like a good time to address it.)

Reply:
There's pros and cons to doing that .. eg, it will make the usage on Mac more annoying - since 99% of the time we are operating on non-Mac packages.  There doesn't seem to be any great solution - even when we get the target to be part of the ident they basically will still have to type in the target.

Reply:
🤔 Hrmm, that's true....

I wonder if it would be worthwhile to allow users to define an environment variable for their commonly-used platform target so it could "do what they mean" in this case.

Reply:
Also, instead of duplicating the same `@arg PKG_TARGET` 4 times, let's add a single function, like for [`arg_cache_key_path`](https://github.com/habitat-sh/habitat/blob/68b1ee1defb9f8b27b54d4e8233f1c707f158fa0/components/hab/src/cli.rs#L837).

Reply:
Thats an interesting proposal @baumanj .. having non-consistent defaults might be the pragmatic option. 

Alternatively, we could also consider enforcing target as a required param.  That way, there would be no confusion or misunderstanding of the user intent, and it would align to what we eventually want to do anyways (make the target mandatory).  The main downside is it would be a breaking change.

Reply:
I like the non-consistent defaults idea, along with an environment variable.

Reply:
Adding the package target option with a function call is also a good call.

Reply:
cool sounds good - will go with the non-consistent option